src/Gui/View3DInventor.cpp

```diff
@@ -513,13 +513,18 @@ void View3DInventor::printPreview()
 {
     QPrinter printer(QPrinter::ScreenResolution);
     printer.setFullPage(true);
-    printer.setPageSize(QPageSize(QPageSize::A4));
-    printer.setPageOrientation(QPageLayout::Landscape);
+    hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/View");
+    int defaultPageSize = hGrp->GetInt("DefaultPageSize", 0); //A4

```

Comment:
Is there a reason here to use a literal zero instead of, e.g. `QPageSize::A4`?

Reply:
That would be better.  I'll do it that way.